bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/WakeOnLanUtility.java

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.lgwebos.internal;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.InterfaceAddress;
+import java.net.NetworkInterface;
+import java.util.Enumeration;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class with utility functions to support Wake On Lan (WOL)
+ *
+ * @author Arjan Mels - Initial contribution
+ * @author Sebastian Prehn - Modification to getMACAddress
+ *
+ */
+@NonNullByDefault
+public class WakeOnLanUtility {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(WakeOnLanUtility.class);
+    private static final Pattern MAC_REGEX = Pattern.compile("(([0-9a-fA-F]{2}[:-]){5}[0-9a-fA-F]{2})");
+
+    /**
+     * Get MAC address for host usesing "arping" tool
+     *
+     * @param hostName Host Name (or IP address) of host to retrieve MAC address for
+     * @return MAC address
+     */
+    public static @Nullable String getMACAddress(String hostName) {
+        try {
+            Process proc = Runtime.getRuntime().exec("arping -r -c 1 -C 1 " + hostName);
```

Comment:
The `arping` utility might not be present on the user's machine (like a windows machine). Is there any way you can make this work on a windows machine as well?

Reply:
that is actually why I changed the code to find the mac address with a regular expression.

just don't know how to detect arp for windows, arp for linux.

Reply:
I found [this](https://stackoverflow.com/questions/20117248/how-to-find-remote-system-mac-address). Looks a little complicated but if it works, it works.

Reply:
yes, the examples show that arp can be used as an alternative. This is what I meant earlier.
on linux `arp <ip>`. On windows `arp -a <ip>`. 

just pushed a new version. I have not tested win and mac as I don't have an OH installation on those envs.

Reply:
Now also tested successfully on my mac

Reply:
It is alright if you can't test on windows. It is a bit unreasonable to expect you to have every kind of environment to test on. I'll accept it as long as you use an approach that the wider internet generally agrees would work.